composeApp/src/commonMain/kotlin/sp/bvantur/tasky/home/data/HomeRepositoryImpl.kt

```diff
@@ -51,4 +51,10 @@ class HomeRepositoryImpl(
             syncScheduler.scheduleAgendaSync(eventsToSync)
         }
     }
+
+    override fun getProfileName(): String? = localDataSource.getProfileName()
+
+    override suspend fun logoutUser(): TaskyEmptyResult<TaskyError> = localDataSource.clearLocalData().onSuccess {
+        remoteDataSource.logoutUser()
```

Comment:
How do you handle errors from the backend in a logout case?

Reply:
Considering Tasky is an offline-first application, I would still allow users to log out when they are offline. With the user logged out and the user’s data cleared in your app, the backend is the only one that doesn’t know that the user is logged out. Depending on how the backend is set up, this may not be a problem, but you can use WorkManager to schedule a logout call.

The above also applies when an error is returned from the logout endpoint, in which case, the odds are that it will be a developer mistake like forgetting to add the API key 😄